lock-api/src/test/java/com/palantir/lock/client/UnreliableTimelockClientTest.java

```diff
@@ -70,19 +78,27 @@ public void verifyIsInitializedCallsDelegate() {
     @Test
     public void verifyGetFreshTimestampCallsDelegate() {
         unreliableTimeLockService.getFreshTimestamp();
-        verify(timeLockClient).getFreshTimestamp();
+        verify(timestampManager).getFreshTimestamp();
     }
 
     @Test
     public void verifyGetCommitTimestampsCallsDelegate() {
         unreliableTimeLockService.getCommitTimestamp(1, LOCK_TOKEN);
-        verify(timeLockClient).getCommitTimestamp(1, LOCK_TOKEN);
+        verify(timestampManager).getCommitTimestamp(1, LOCK_TOKEN);
     }
 
     @Test
     public void verifyGetFreshTimestampsCallsDelegate() {
         unreliableTimeLockService.getFreshTimestamps(1);
-        verify(timeLockClient).getFreshTimestamps(1);
+        verify(timestampManager).getFreshTimestamps(1);
+    }
+
+    @ParameterizedTest
+    @MethodSource("timestampMethods")
+    public void timestampMethodsRandomlyIncreaseTimestamp(Consumer<UnreliableTimeLockService> task) {
```

Comment:
consider: testing this for a case where the buggify is always false and/or (better) use an implementation that is non-degenerate (i.e. test true/false in the same test)

Reply:
There are already tests! Strengthened the existing ones, though